core/src/main/java/com/novoda/merlin/receiver/ConnectivityCallbacks.java

```diff
@@ -0,0 +1,50 @@
+package com.novoda.merlin.receiver;
+
+import android.annotation.TargetApi;
+import android.net.ConnectivityManager;
+import android.net.Network;
+import android.net.NetworkInfo;
+import android.os.Build;
+
+import com.novoda.merlin.service.MerlinService;
+
+@TargetApi(Build.VERSION_CODES.LOLLIPOP)
+public class ConnectivityCallbacks extends ConnectivityManager.NetworkCallback {
+
+    private final ConnectivityManager connectivityManager;
+    private final MerlinService merlinService;
```

Comment:
@Mecharyry I think that unbind is just called from [Merlin:unbind](https://github.com/novoda/merlin/blob/ba9cd74aa7d2b1a295a9a6867ca41bb55409019e/core/src/main/java/com/novoda/merlin/Merlin.java#L34-L36) As far as I've seen in the docs, I think that [MerlinServiceBinder.Connection:onServiceDisconected](https://github.com/novoda/merlin/blob/master/core/src/main/java/com/novoda/merlin/service/MerlinServiceBinder.java#L86-L88) is what is called if the service is killed. What I don't really know if it's doing a proper cleanup there. May it be possible to do some cleaning in MerlinService.onDestroy?